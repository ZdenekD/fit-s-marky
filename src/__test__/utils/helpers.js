import React from 'react';
import {shallow} from 'enzyme';

/**
 * Factory function to create a shallow wrapper for App component
 * @function setup
 * @param {ReactComponent} Component - React component to render with shallow
 * @param {Object} props - Component props specific to this setup
 * @returns {ShallowWrapper}
 */
export const factory = (Component, props = {}) => {
    const wrapper = shallow(<Component {...props} />);

    return wrapper;
};

/**
 * Return ShallowWrapper containg node(s) with the data-test value
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {string} value - data attribute value to find
 * @returns {ShallowWrapper}
 */
export const findComponent = (wrapper, value) => wrapper.find(`[data-test="${value}"]`);
