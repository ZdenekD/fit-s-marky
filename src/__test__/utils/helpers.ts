import {ShallowWrapper} from 'enzyme';

const findComponent = (wrapper: ShallowWrapper, value: string): ShallowWrapper => wrapper.find(`[data-test="${value}"]`);

export default findComponent;
