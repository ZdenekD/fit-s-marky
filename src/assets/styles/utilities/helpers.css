:global ._pull-right {
    float: right !important;
}

:global ._pull-left {
    float: left !important;
}

:global ._italic {
    font-style: italic;
}
