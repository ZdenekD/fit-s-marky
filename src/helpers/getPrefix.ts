const getPrefix = (): string => '_5innv';

export default getPrefix;
