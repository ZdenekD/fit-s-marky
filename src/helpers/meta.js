import React from 'react';

const getMeta = item => (
    [
        <meta name="description" content={item.description} key={item.id} />,
        <meta property="og:title" content={item.title} key={item.id} />,
        <meta property="og:type" content="website" key={item.id} />,
        <meta property="og:description" content={item.description} key={item.id} />,
        <meta property="og:url" content="https://fit-s-marky.cz/" key={item.id} />,
        <meta name="twitter:card" content="summary" key={item.id} />,
        <meta name="twitter:title" content={item.title} key={item.id} />,
        <meta name="twitter:description" content={item.description} key={item.id} />,
        <meta name="twitter:domain" content="https://fit-s-marky.cz/" key={item.id} />,
    ]
);

export default getMeta;
