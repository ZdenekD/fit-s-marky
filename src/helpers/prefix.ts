const prefix = '_5innv';

export default prefix;
