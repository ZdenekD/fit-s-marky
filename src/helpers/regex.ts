interface IRegex {
    text: RegExp
    zip: RegExp
    email: RegExp
}

const regex: IRegex = {
    text: /^[éěëščřžýáäíïťďňöúůüÿÉĚËŠČŘŽÝÁÄÍÏŤĎŇÖÚŮÜŸ \-,.?!'"\w\s\d]+$/,
    zip: /^\d{3}(\s)?\d{2}$/,
    email: /^("([ !#-[\]-~]|\\[ -~])+"|[-a-z0-9!#$%&'*+/=?^_`{|}~]+(\.[-a-z0-9!#$%&'*+/=?^_`{|}~]+)*)@([0-9a-z\u00C0-\u02FF\u0370-\u1EFF]([-0-9a-z\u00C0-\u02FF\u0370-\u1EFF]{0,61}[0-9a-z\u00C0-\u02FF\u0370-\u1EFF])?\.)+[a-z\u00C0-\u02FF\u0370-\u1EFF]([-0-9a-z\u00C0-\u02FF\u0370-\u1EFF]{0,17}[a-z\u00C0-\u02FF\u0370-\u1EFF])?$/i,
};

export default regex;
