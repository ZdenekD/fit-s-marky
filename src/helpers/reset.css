a,
abbr,
address,
article,
aside,
audio,
b,
blockquote,
body,
button,
canvas,
caption,
cite,
code,
dd,
del,
details,
dfn,
div,
dl,
dt,
em,
embed,
fieldset,
figcaption,
figure,
footer,
form,
h1,
h2,
h3,
h4,
h5,
h6,
header,
html,
i,
iframe,
img,
input,
ins,
kbd,
label,
legend,
li,
main,
mark,
nav,
object,
ol,
output,
p,
pre,
q,
ruby,
s,
samp,
section,
select,
small,
span,
strong,
sub,
summary,
sup,
table,
tbody,
td,
textarea,
tfoot,
th,
thead,
time,
tr,
u,
ul,
var,
video {
    margin: 0;
    padding: 0;
    vertical-align: baseline;
    background: transparent;
    font: inherit;
    font-size: 100%;
    outline: 0;
    border: none;
}

*,
*::after,
*::before {
    box-sizing: border-box;
}

body {
    height: 100%;
    overflow-y: scroll;
    line-height: 1;
}

article,
aside,
details,
figcaption,
figure,
footer,
header,
main,
nav,
section {
    display: block;
}

audio,
canvas,
video {
    display: inline-block;
}

ol,
ul {
    list-style: none;
}
