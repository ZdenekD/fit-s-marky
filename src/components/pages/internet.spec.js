import {factory} from '../../__test__/utils/helpers';
import Internet from './internet';

describe('Internet page', () => {
    it('match snapshot', () => {
        const wrapper = factory(Internet);

        expect(wrapper.html()).toMatchSnapshot();
    });
});
