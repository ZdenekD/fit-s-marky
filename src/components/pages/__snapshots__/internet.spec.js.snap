// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Internet page match snapshot 1`] = `
<header>
  <h2>
    Osobní jídelníček po internetu
  </h2>
</header>
<h3>
  Jídelníček je určen pro Vás, kteří chcete:
</h3>
<ul>
  <li>
    sestavit jídelníček výživovým poradcem po internetu
  </li>
  <li>
    chcete si nechat zkontrolovat současný jídelníček odborníkem
  </li>
  <li>
    chcete dostat několik užitečných nápadů, jak lze jídelníček rozšířit, zpestřit či doplnit
  </li>
</ul>
<h3>
  Jak vypadá jídelníček?
</h3>
<p>
  Jídelníček sestavuji na základě podrobného dotazníku. Dotazník zahrnuje otázky týkající se Vašeho současného jídelníčku, zdravotního stavu, pohybového režimu a dalších detailů. Dotazník vyplníte on-line, odešlete a na Vaši e-mailovou adresu obdržíte hotový jídelníček, který má několik částí:
</p>
<ul>
  <li>
    analýzu a zhodnocení Vašich současných stravovacích návyků, které je nutné změnit a také ty, které si můžete ponechat
  </li>
  <li>
    plán a doporučení pro Vaši redukci či udržení váhy
  </li>
  <li>
    potraviny, které nejsou do jídelníčku vhodné zařazovat a proč
  </li>
  <li>
    seznam pravidel a doporučení pro dosažení Vašeho cíle
  </li>
  <li>
    úprava a doporučení týkající se pitného režimu
  </li>
  <li>
    příklad vhodných jídel – několik tipů na zdravý oběd, snídani …
  </li>
</ul>
<p>
  <strong>
    Jídelníček neobsahuje přesný popis každodenních jídel (např. ráno 200g jogurtu a müsli …), ale jde o praktický návod jak nahradit nevhodné potraviny zdravějšími, jakou volit úpravu jídel, jak jíst v případě vytíženosti v zaměstnání …
  </strong>
</p>
`;
